### PR TITLE
Upload data

### PR DESCRIPTION
**Pull request by sedroche**
# Motivation
Ticket: https://issues.jboss.org/browse/RHMAP-11591

We need to allow uploading of collections from json, csv and bson files

### CODE REVIEW

**Review by rachael-oregan — APPROVED** [review]
LGTM

File: src/endpoints/http/collections/index.js [hunk 1]
```diff
@@ -1,6 +1,14 @@
 import listCollections from './list';
 import createCollection from './create';
 import deleteCollections from './delete';
+import {insertCollection, getCollectionName} from './files';
+import parseFile from '../../../middleware/parse-file';
+
+const CREATED = 201;
```
Comment (pb82) [inline src/endpoints/http/collections/index.js]: Should we move all of those code into one reusable file?

File: src/middleware/parse-file/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import Busboy from 'busboy';
+import parsers from './parsers';
+
+const  UNSUPPORTED_MEDIA = 415;
```
Comment (pb82) [inline src/middleware/parse-file/index.js]: see above comment

File: src/endpoints/http/collections/files/mongoStream/InsertMixin.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * Mixin implementing the WriteStream interface.
+ * InsertMixin will insert object(s) into a mongodb collection.
+ *
+ */
+var InsertMixin = Base => class extends Base {
+
+  constructor(options={}) {
+    if (!options.db || !options.collectionName) {
```
Comment (pb82) [inline src/endpoints/http/collections/files/mongoStream/InsertMixin.js:27]: maybe user asserts here?

**Review by pb82 — APPROVED** [review]
LGTM, just some comments

**Pull request merged**